bitcoin/dogeblock.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bitcoin
+
+import (
+	"encoding/binary"
+	"io"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	vChainID       = 0x0062
```

Comment:
This is a Dogecoin specific configuration (and the only one in the entire file.) We should make this variable - maybe add a field to the Block struct we define here? This standard implementation for AuxPoW always requires a ChainId anyway.

Reply:
We could pass it as an argument to the block decoder. Should the method continue to decode auxpow and non-auxpow blocks? The current implementation assumes it's a dogecoin block which could be either auxpow or non-auxpow.

Reply:
Yes. Most coins have that implementation.

Reply:
Should it return an error if the AuxPoW flag is set but the Chain ID doesn't match? If not, how should it decode the block?

Reply:
Yeah this should never happen behind an actively validating dogecoind (or namecoin node for example.) So I would just throw an error in that case.

Reply:
We could also ignore the Chain ID and just check if the AuxPoW flag is set if we don't need to validate the Chain ID.